packages/tools/src/tools/annotation/planarFreehandROITool/renderMethods.ts

```diff
@@ -0,0 +1,300 @@
+import {
+  drawHandles as drawHandlesSvg,
+  drawPolyline as drawPolylineSvg,
+} from '../../../drawingSvg';
+import { polyline } from '../../../utilities/math';
+import { Settings } from '@cornerstonejs/core';
+import { PlanarFreehandROIAnnotation } from '../../../types/ToolSpecificAnnotationTypes';
+import type { Types } from '@cornerstonejs/core';
+
+const { pointsAreWithinCloseContourProximity } = polyline;
+
+/**
+ * Renders a `PlanarFreehandROIAnnotation` that is not currently being drawn or edited.
+ */
+function renderContour(
+  enabledElement: Types.IEnabledElement,
+  svgDrawingHelper: any,
+  annotation: PlanarFreehandROIAnnotation
+): void {
+  if (annotation.data.isOpenContour) {
+    this.renderOpenContour(enabledElement, svgDrawingHelper, annotation);
+  } else {
+    this.renderClosedContour(enabledElement, svgDrawingHelper, annotation);
+  }
+}
+
+/**
+ * Renders an closed `PlanarFreehandROIAnnotation` annotation.
+ */
+function renderClosedContour(
+  enabledElement: Types.IEnabledElement,
+  svgDrawingHelper: any,
+  annotation: PlanarFreehandROIAnnotation
+): void {
+  const { viewport } = enabledElement;
+
+  const settings = Settings.getObjectSettings(annotation, this.getToolName());
+
+  // Todo -> Its unfortunate that we have to do this for each annotation,
+  // Even if its unchanged. Perhaps we should cache canvas points per element
+  // on the tool? That feels very weird also as we'd need to manage it/clean
+  // them up.
+  const canvasPoints = annotation.data.polyline.map((worldPos) =>
```

Comment:
Yes I agree, this was not problem in other annotation tools ....

Reply:
This is comment for the future. This will happen for any significantly large annotation. Its an optimisation we don't have time to tackle here yet.